mempool/src/shared_mempool/tasks.rs

```diff
@@ -215,17 +212,15 @@ where
             })
             .collect();
 
-    let validation_results = join_all(transactions.iter().map(|t| {
-        let vm_validator = smp.validator.clone();
-        async move {
-            vm_validator
+    let validation_results = transactions
```

Comment:
wow! this looks bad
so we are validating transactions one by one now instead of doing it in parallel
but probably gRPC was so bad that it matters much less:) 
maybe we need to change vm_validator interface cc: @christinacelee

Reply:
I didn't notice the perf in this aspect but it seems What gRPC brings here is just threadpool. I think we also discussed async-storage api before. If storage IO really becomes a bottleneck maybe we will add async support.
cc: @msmouse

Reply:
from what I remember from @frumious insights, bottleneck was vm validation itself. That's why we parallelize it here. Would be interesting to take a closer look again. But anyway, your PR actually improves perf, so we can follow up on that later